Street Trees/STUserDetailsFormViewController.swift

```diff
@@ -0,0 +1,130 @@
+    func contactDetailsFormViewController(form: STContactDetailsFormViewController, didCompleteWithContact aContact: STTKContact)
+}
+
+//**********************************************************************************************************************
+// MARK: - Class Implementation
+
+class STContactDetailsFormViewController: UIViewController {
+    
+    var activeTextField: UITextField?
+    var address: STTKStreetAddress?
+    
+    weak var delegate: STContactDetailsFormViewControllerDelegate?
+    
+    @IBOutlet weak var nameTextField: UITextField!
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var phoneNumberTextField: UITextField! {
+        didSet {
+            self.phoneNumberTextField.inputAccessoryView = self.toolbar
+        }
+    }
+    @IBOutlet weak var toolbar: UIToolbar!
+    
+
+    //******************************************************************************************************************
+    // MARK: - TextField Delegate
+    
+    func textFieldDidBeginEditing(textField: UITextField) {
+        self.activeTextField = textField
+    }
+    
+    func textFieldDidEndEditing(textField: UITextField) {
+        self.activeTextField = nil
+    }
+    
+    func textFieldShouldReturn(textField: UITextField) -> Bool {
+        
+        switch textField {
+        case self.nameTextField:
+            self.emailTextField.becomeFirstResponder()
+            return false
+        case self.emailTextField:
+            self.phoneNumberTextField.becomeFirstResponder()
+            return false
+        case self.phoneNumberTextField:
+            fallthrough
+        default:
+            return true
+        }
+    }
+    
+    //******************************************************************************************************************
+    // MARK: - Actions
+    
+    @IBAction func dismissKeyboard(sender: AnyObject) {
+        self.nameTextField.resignFirstResponder()
+        self.emailTextField.resignFirstResponder()
+        self.phoneNumberTextField.resignFirstResponder()
+    }
+    
+    @IBAction func nextButton(sender: UIButton) {
+        
+        guard let name = self.nameTextField.text where !name.isEmpty else {
+            self.showAlert(STContactErrorTitle, message: STContactErrorNameMessage)
+            return
+        }
+        
+        guard let email = self.emailTextField.text where self.validate(string: email, validation: .Email) else {
+            self.showAlert(STContactErrorTitle, message: STContactErrorEmailMessage)
+            return
+        }
+        
+        guard let phoneNumber = self.phoneNumberTextField.text where self.validate(string: phoneNumber, validation: .PhoneNumber) else {
+            self.showAlert(STContactErrorTitle, message: STContactErrorNumberMessage)
+            return
+        }
+        
+        guard let safeAddress = self.address else {
+            fatalError("An Address has not been passed.")
```

Comment:
I choose fatal error so that when tested, this will cause a crash and the person implementing the view controllers use won't forget to send in the address object.

It is pretty dangerous I agree, however I feel that it is required because the form cannot be complete if the developer forgets to pass in the address object.